### DISCUSSION TITLE
[React Native] auth.signOut unhandled promise rejection.

### DISCUSSION THREAD

**Discussion by kirkkaf13** [supabase/supabase#5058] [Questions] [answered]
Hello, 

I have a expo react native application and have called the auth.signOut() function. It appears signout just fine according to the audit trail but I am experiencing a warning relating to an unhandled promise rejection and I can't seem to track down the issue.

Does anyone have any experience using signOut() in a RN application or have an example I could look at?

Update: The stack trace seems to point at the removeAllSubscriptions() in the SupabaseClient.js.

Thank you.

**Comment by GaryAustin1** [comment, marked as answer]
I was involved in a bug Supabase fixed that modified that function about a month ago.
https://github.com/supabase/supabase-js/commit/8253b773fd6ab2ced84ef27d80bdc80a344d54c8
Are you using realtime subscriptions?

The function was changed to async so maybe a change needed to occur in another Supabase function calling it, or it is getting an error that is not really expected... 
@w3b6x9 you might want to be aware of this...

**Reply by kirkkaf13** [reply]
Firstly thanks for making me aware of these changes. After further investigation I don't believe this was the problem in my case. I looked at my console log of my IDE rather than my device and it appears it's a type  issue.

> Unhandled promise rejection: TypeError: Promise.allSettled is not a function.

I suspect that Promise isn't available in a Expo RN application and it needs a polyfill.

**Reply by w3b6x9** [reply]
@GaryAustin1 thanks for bringing this to my attention!

@kirkkaf13 i'll go and refactor the `removeAllSubscriptions` function so it doesn't call `Promise.allSettled`.

**Reply by kirkkaf13** [reply]
@w3b6x9 Excellent thank you. For now I have fixed it in my app by polyfilling with Promise.all().

**Reply by w3b6x9** [reply]
@kirkkaf13 can you please take a look at this [PR](https://github.com/supabase/supabase-js/pull/344) and/or testing it locally to make sure it works for you?
